File: src/playasophy/wonderdome/input/file_system_handler.clj
(ns playasophy.wonderdome.input.file-system-handler
  (:gen-class
    :name playasophy.wonderdome.input.FileSystemHandler
    :state root
    :init init
    :constructors
    {[String] []}
    :methods
    [[sketchPath [String] String]
     [createInput [String] java.io.InputStream]]))


(defn -init
  [root]
  [[] root])


(defn -sketchPath
  ^String
  [this file-name]
  (str (.root this) \/ file-name))


(defn -createInput
  [this file-name]
  (java.io.FileInputStream. (-sketchPath this file-name)))

File: src/java/playasophy/wonderdome/input/FileSystemHandler.java
package playasophy.wonderdome.input;


import java.io.FileInputStream;
import java.io.InputStream;


public class FileSystemHandler {

    public final String root;


    public FileSystemHandler(String root) {
        this.root = root;
    }


    public String sketchPath(String fileName) {
        return root + "/" + fileName;
    }


    public InputStream createInput(String fileName) throws Exception {
        return new FileInputStream(sketchPath(fileName));
    }

}
